src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -40,8 +42,9 @@ internal TestDiagnosticAnalyzerService(ImmutableArray<AnalyzerReference> hostAna
             HostAnalyzerManager hostAnalyzerManager,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
             Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException,
-            IDiagnosticUpdateSourceRegistrationService registrationService = null)
-            : base(hostAnalyzerManager, hostDiagnosticUpdateSource, registrationService ?? new MockDiagnosticUpdateSourceRegistrationService())
+            IDiagnosticUpdateSourceRegistrationService registrationService = null,
+            IAsynchronousOperationListener listener = null)
```

Comment:
I'm worried we may end up with flakeyness again in the future when something else forgets to pass this along.  It could be non-optional but tests could still pass in null to explicitly state they didn't need a listener.  But then tests would at least have to think about it and document that that was ok for them.